components/brave_wallet/browser/keyring_service.cc

```diff
@@ -1314,16 +1328,18 @@ void KeyringService::IsLocked(IsLockedCallback callback) {
   std::move(callback).Run(IsLocked());
 }
 
-void KeyringService::Reset(bool notify_observer) {
+void KeyringService::Reset(const std::string& keyring_id,
+                           bool notify_observer) {
   StopAutoLockTimer();
-  encryptor_.reset();
-  keyrings_.clear();
-
-  ClearKeyringServiceProfilePrefs(prefs_);
+  encryptors_.erase(keyring_id);
+  keyrings_.erase(keyring_id);
+  if (keyring_id == mojom::kDefaultKeyringId) {
+    ClearKeyringServiceProfilePrefs(prefs_);
```

Comment:
Reset with keyring_id is problematic because this will clear `kBraveWalletKeyrings` when keyring is default, so we also need to clear all encryptors_ and keyrings_ because other keyring's prefs are cleared too. That being said, default keyring must exists for other types of keyrings. 
I think it is better to leave it unchanged in this PR and we can open a separate issue to address it.

Reply:
issue opened here
https://github.com/brave/brave-browser/issues/20731

Reply:
done